src/pages/Scholarships.jsx

```diff
@@ -27,13 +30,57 @@ function ScholarshipsPage() {
   const [sortField, setSortField] = useState('deadline');
   const [sortDir, setSortDir] = useState('asc');
 
+  const location = useLocation();
+  const history = useHistory();
+
+  // sets min and max amount for the amount filter
+  const setMinMax = (index, val) => {
+    history.push({
+      search: queryString.stringify({
+        ...queryString.parse(location.search),
+        [index]: val,
+      }),
+    });
+  };
+
+  const params = queryString.parse(location.search, { parseNumbers: true });
+  const minParam = params.minAmount;
+  const maxParam = params.maxAmount;
+
+  if ((minParam && !Number.isInteger(minParam)) || minParam === 0) {
+    delete params.minAmount;
+    history.replace({ search: queryString.stringify(params) });
+  }
+
+  if ((maxParam && !Number.isInteger(maxParam)) || maxParam === 0) {
+    delete params.maxAmount;
+    history.replace({ search: queryString.stringify(params) });
+  }
+
   useEffect(() => {
     // TODO: Create cancellable promises
     Scholarships.list({ sortField, sortDir })
-      .then(setScholarships)
+      .then((results) => {
+        const filterError = maxParam > 0 && maxParam <= minParam;
+
+        if (filterError) setScholarships(results);
+        else
+          setScholarships(
+            results.filter(({ data }) => {
+              if (data.amount.type === AmountType.Unknown) return true;
```

Comment:
Without this, the Scholarships with Unknown amount don't show.

Reply:
Why is that? Scholarships with Unknown amount have a max of -1 (so `data.amount.max < minParam` should always be true) and a min of over a billion (so `data.amount.min > maxParam` should always be true).

I wonder if it's just the UNKNOWN scholarship data you have in your local firestore emulator. Try creating a new UNKNOWN scholarship and see if it shows up.

Reply:
The reason was because both conditions are true and therefore returns false.